lib/mapping/CMapDefines.h

```diff
@@ -82,11 +82,11 @@ struct DLL_LINKAGE TerrainTile
 	EDiggingStatus getDiggingStatus(const bool excludeTop = true) const;
 	bool hasFavorableWinds() const;
 
-	Terrain terType;
+	TerrainType * terType;

```

Comment:
Is it possible to store Id instead of pointers? For multiplayer we need to transfer the map over network and it's tough with pointers

Reply:
Will try and see how it performs

Reply:
For some reason, accessing terrains via id / VLC is slower:

<html>
<body>


+HoTA new | Terrain by ptr | Terrain by id
-- | -- | --
objects | 3493 | 4034
  | 3626 | 4135
  | 3472 | 3980
  | 3829 | 3948
  | 3538 | 3737
  | 3630 | 3776
  | 3603 | 3740
  | 3807 | 3852
  | 4089 | 3739
  | 3844 | 3895
mean time (ms) | 3693 | 3884
Time % |   | 105,16%
Improvement |   | -5,16%


</body>

</html>

Reply:
Give it a try: 
[handle_terrain_by_id_patch.zip](https://github.com/vcmi/vcmi/files/9662184/handle_terrain_by_id_patch.zip)

Reply:
Is something like that possible:

```
if (h.saving)
{
auto id =  terType.id;
h & id;
}
else
{
h & id;
terType = &(VLC->terrainTypeHandler->terrains()[id]);
}
```

Reply:
I think it should work

Reply:
Btw, passing map over network speeds up rmg twice as well :)